src/io/mod.rs

```diff
@@ -0,0 +1,13 @@
+#[cfg(target_os = "linux")]
+mod unix;
+#[cfg(target_os = "linux")]
+pub(crate) use unix::File;
+#[cfg(target_os = "linux")]
+pub use unix::IODriver;
+
+#[cfg(all(target_os = "macos", not(feature = "async-io-rio")))]
+mod linux;
+#[cfg(all(target_os = "macos", not(feature = "async-io-rio")))]
+pub(crate) use unix::File;
+#[cfg(all(target_os = "macos", not(feature = "async-io-rio")))]
+pub use unix::IODriver;
```

Comment:
On this level declarations are identical, so no need to split them for `linux` and `macos`. It is better to use `#[cfg(target_family = "unix")]` and unify declarations